### DISCUSSION TITLE
I Can't use bluetooth methods.

### DISCUSSION THREAD

**Discussion by avcismail** [kenjdavidson/react-native-bluetooth-classic#312] [Q&A]
Hello everyone,

I include this library in my project, but I cannot use the library methods. I set permissions in AndroidManifest file. Do I need to set permissions for iOS? For example, I am trying to use the isBluetoothEnabled() method, but it falls into the catch block within the try catch structure.

` LOG  [TypeError: Cannot read property 'isBluetoothEnabled' of null]`

My code ise here:

```
import React, { useState, useEffect } from "react";
   import { AppRegistry, StyleSheet, Text, View, Platform } from "react-native";
   import RNBluetoothClassic, {
   BluetoothEventType,
  BluetoothDevice,
  requestBluetoothEnabled
} from "react-native-bluetooth-classic";


export default function App() {
  const [enabled, setEnabled] = useState(false);
    const [connected, setConnected] = useState(false);
    const [bluetooth, setBluetooth] = useState(false);
    const deviceMac = "0C:B8:15:75:71:6A";
    useEffect(() => {
      checkBluetoothEnabled();
    }, []);

    const checkBluetoothEnabled = async () => {
      try {
        const isEnabled = await RNBluetoothClassic.isBluetoothEnabled();
        const paired = await RNBluetoothClassic.getBondedDevices();

        setEnabled(isEnabled)
    
        console.log("Bluetooth enabled: " + enabled);
        console.log("Bluetooth Connected: " + paired);
      } catch (error) {
        console.log(error);
      }
    };

  return (
    <View style={styles.container}>
      <Text>Bluetooth App</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
});

AppRegistry.registerComponent("App", () => App);
```

I've been searching for days but I haven't found any results. Can you help me?

**Comment by kenjdavidson** [comment]
So on Android it works but ios it doesn't? Sounds like it's hasn't installed properly on ios. Can you try manually running the podspec? Or try reinstalling the library?

**Reply by avcismail** [reply]
I couldn't run it on Android either. It may be caused by the emulator. I do not have a physical Android device. I reinstalled the library a few times. I even deleted the node_modules folder and did npm install again, but it didn't work. I'll try running podspec manually.

**Comment by kenjdavidson** [comment]
Not sure to be honest. I've never used the emulator, so I can't guarantee that Bluetooth is available on it.

**Reply by avcismail** [reply]
Yes, I cannot guarantee whether it will work in the emulator or not. However, if I can run it on iOS, I will also test it on Android with a physical device.

**Reply by avcismail** [reply]
I tried to install it manually with pod install but it still didn't work. There is a topic in your document under the heading MFi Protocols. Do we need to make any settings regarding this issue?

**Reply by kenjdavidson** [reply]
no clue, without more of a stack trace or you debugging some for me, it's pretty hard to say just from the error you've provided above.

- what version are you using of android and ios
- what version of the library
- what version of react native

you need to provide a bit more information.

**Reply by avcismail** [reply]
Okey, here is `react-native info`

```
System:
  OS: macOS 13.6.4
  CPU: (4) x64 Intel(R) Core(TM) i7-7567U CPU @ 3.50GHz
  Memory: 1.34 GB / 16.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.11.1
    path: /usr/local/bin/node
  Yarn:
    version: 1.22.21
    path: /usr/local/bin/yarn
  npm:
    version: 10.4.0
    path: /usr/local/bin/npm
  Watchman:
    version: 2024.01.22.00
    path: /usr/local/bin/watchman
Managers:
  CocoaPods:
    version: 1.15.2
    path: /usr/local/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.2
      - iOS 17.2
      - macOS 14.2
      - tvOS 17.2
      - visionOS 1.0
      - watchOS 10.2
  Android SDK:
    API Levels:
      - "33"
      - "34"
    Build Tools:
      - 29.0.2
      - 30.0.3
      - 33.0.0
      - 33.0.1
      - 34.0.0
    System Images:
      - android-30 | Google Play Intel x86 Atom
      - android-34 | Google APIs Intel x86_64 Atom
    Android NDK: Not Found
IDEs:
  Android Studio: 2022.3 AI-223.8836.35.2231.10671973
  Xcode:
    version: 15.2/15C500b
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 21.0.1
    path: /usr/bin/javac
  Ruby:
    version: 2.6.10
    path: /usr/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.73.4
    wanted: 0.73.4
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: false

info React Native v0.73.5 is now available (your project is running on v0.73.4).
```

**Reply by kenjdavidson** [reply]
Looks fine, the fact that there aren't a bunch of open issues with these versions would lead me to believe it's an issue with your project at the moment.  I would suggest looking through the open issues to see if something matches the error you're seeing (there have been, so maybe there is a resolution that works for you).

But realistically not much i can do without the project, which, I probably wouldn't have time to setup and debug for you anyhow.  If you debug it and find something, I'll definitely assist in working through resolving it.

**Reply by kenjdavidson** [reply]
I would suggest looking at the Logcat of the Android studio and see if there are any native side errors which would be stopping the NativeModule from loading into Javascript.

**Reply by kenjdavidson** [reply]
The only major issue is that if React Native finally removed this type of native module.  In which case I would suggest reverting to an earlier version of react native as I probably won't get around to converting this to a Turbo module.

**Reply by avcismail** [reply]
Okay, I hope I can find more error details. Thank you for your help.

**Reply by wizzer129** [reply]
Hey were you able to discover what was causing this issue for you? I'm experiencing something similar when executing any of the functions in RNBluetoothClassic on my device.

**Reply by avcismail** [reply]
Hello, no I couldn't find it. So I started using a different package. But I haven't tested the iOS side yet. @wizzer129

**Comment by kenjdavidson** [comment]
I would open an issue, but it sound slike you've got it working with a different library.
